/* Fix issue with OSX causing "Invalid response packet from host" messages. */
--- a/avahi-core/server.c	2012-02-27 14:22:41.000000000 -0800
+++ b/avahi-core/server.c	2012-03-12 16:50:34.675656283 -0700
@@ -979,7 +979,8 @@
         }
 
         if (avahi_dns_packet_get_field(p, AVAHI_DNS_FIELD_QDCOUNT) != 0 ||
-            avahi_dns_packet_get_field(p, AVAHI_DNS_FIELD_ANCOUNT) == 0 ||
+            (avahi_dns_packet_get_field(p, AVAHI_DNS_FIELD_ANCOUNT) +
+             avahi_dns_packet_get_field(p, AVAHI_DNS_FIELD_ARCOUNT)) == 0 ||
             avahi_dns_packet_get_field(p, AVAHI_DNS_FIELD_NSCOUNT) != 0) {
 
             avahi_log_warn("Invalid response packet from host %s.", avahi_address_snprint(t, sizeof(t), src_address));
